chain/chunks/src/lib.rs

```diff
@@ -1561,6 +1559,28 @@ impl ShardsManager {
         Ok(())
     }
 
+    /// Processes the forwarded chunk parts for the header, and if the chunk is complete,
+    /// processes the chunk.
+    pub fn process_cached_chunk_forwards_for_header(
+        &mut self,
+        header: &ShardChunkHeader,
+        chain_store: &mut ChainStore,
+        rs: &mut ReedSolomonWrapper,
+    ) -> Result<ProcessPartialEncodedChunkResult, Error> {
+        if self.encoded_chunks.get_or_insert_from_header(header).complete {
+            return Ok(ProcessPartialEncodedChunkResult::Known);
+        }
+        if let Some(parts) = self.chunk_forwards_cache.pop(&header.chunk_hash()) {
+            self.encoded_chunks.merge_in_partial_encoded_chunk(&PartialEncodedChunkV2 {
+                header: header.clone(),
+                parts: parts.into_values().collect(),
+                receipts: vec![],
+            });
+        }
+
+        self.try_process_chunk_parts_and_receipts(header, chain_store, rs)
```

Comment:
I think code readability might be better if we move this line outside of this function. It's not obvious from the function name that this function also tries to process chunk parts and receipts.

Reply:
Moved out.